src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonResource.kt

```diff
@@ -105,6 +106,23 @@ class PrisonResource(private val prisonService: PrisonService, private val addre
     @Parameter(description = "Genders to filter by", example = "MALE, FEMALE", required = false) @RequestParam genders: List<Gender>? = listOf(),
     @Parameter(description = "Prison type codes to filter by", example = "HMP, YOI", required = false) @RequestParam prisonTypeCodes: List<Type>? = listOf(),
   ): List<PrisonDto> = prisonService.findByPrisonFilter(active, textSearch, genders, prisonTypeCodes)
+
+  @GetMapping(
+    "/names",
+    produces = [MediaType.APPLICATION_JSON_VALUE],
+  )
+  @Operation(summary = "Get prison names", description = "prison id and full name")
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Successful Operation",
+      ),
+    ],
+  )
+  fun getPrisonNames(): List<PrisonNameDto> {
```

Comment:
why not use the existing /prisons endpoint to get prison names instead of adding a new one?

Reply:
Thats the point of the code, it's a big return.